metadata-service/factories/src/main/java/com/linkedin/restli/server/RAPServletFactory.java

```diff
@@ -0,0 +1,53 @@
+package com.linkedin.restli.server;
+
+import com.linkedin.metadata.filter.RestliLoggingFilter;
+import com.linkedin.parseq.Engine;
+import com.linkedin.parseq.EngineBuilder;
+import com.linkedin.r2.filter.FilterChains;
+import com.linkedin.r2.filter.transport.FilterChainDispatcher;
+import com.linkedin.r2.transport.http.server.RAPServlet;
+import com.linkedin.restli.docgen.DefaultDocumentationRequestHandler;
+import com.linkedin.restli.server.spring.SpringInjectResourceFactory;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.concurrent.Executors;
+
+@Slf4j
+@Configuration
+public class RAPServletFactory {
+    @Value("#{systemEnvironment['RESTLI_SERVLET_THREADS']}")
```

Comment:
Why specifically systemEnvironment rather than letting SPEL decide which one to use? Are we worried this might get overridden incorrectly?

Reply:
My intent here is that really we should never override the default calculation based on the CPU cores per the official guide. This is a parameter that should follow the pod configuration in general and not be modified, therefore I avoided anything related to the application.yaml